command/var_put_test.go

```diff
@@ -107,6 +108,44 @@ func TestVarPutCommand_GoodJson(t *testing.T) {
 	require.Equal(t, api.VariableItems{"k1": "v1", "k2": "v2"}, outVar.Items)
 }
 
+func TestVarPutCommand_FlagsWithSpec(t *testing.T) {
+	ci.Parallel(t)
+
+	// Create a server
+	srv, client, url := testServer(t, true, nil)
+	defer srv.Shutdown()
+
+	ui := cli.NewMockUi()
+	cmd := &VarPutCommand{Meta: Meta{Ui: ui}}
+
+	// Create a temporary file and ensure the proper cleanup is run once the
+	// test ends.
+	osFile, err := os.CreateTemp("", "nomad-cli-var-put-test-*.hcl")
+	must.NoError(t, err)
+
+	t.Cleanup(func() {
+		_ = osFile.Close()
+		_ = os.Remove(osFile.Name())
+	})
+
+	// Write out a partial spec that includes the namespace and variable path.
+	_, err = osFile.Write([]byte("path      = \"path/to/variable\"\nnamespace = \"default\""))
+	must.NoError(t, err)
+
+	// Create the variables, ensure we clean it up, and check the command
+	// response.
+	code := cmd.Run([]string{"-address=" + url, "@" + osFile.Name(), "k1=v1", "k2=v2"})
+	must.Zero(t, code)
+
+	t.Cleanup(func() {
+		_, _ = client.Variables().Delete("test/var", nil)
+	})
```

Comment:
We don't need to worry about this, as we'll be throwing away the in-memory test server at the end of the test.